public/js/controllers/blocklist_controller.js

```diff
@@ -28,4 +35,70 @@ export default class extends Controller {
     url.set('pathname', controller.listviewTarget.selectedOptions[0].value)
     Turbolinks.visit(url.href)
   }
+
+  _processBlock (blockData) {
+    if (!this.hasTableTarget) return
+    var block = blockData.block
+    // Grab a copy of the first row.
+    var rows = this.tableTarget.querySelectorAll('tr')
+    if (rows.length === 0) return
+    var tr = rows[0]
+    var lastHeight = parseInt(tr.dataset.height)
+    // Make sure this block belongs on the top of this table.
+    if (block.height === lastHeight) {
+      this.tableTarget.removeChild(tr)
+    } else if (block.height === lastHeight + 1) {
+      this.tableTarget.removeChild(rows[rows.length - 1])
+    } else return
+    // Set the td contents based on the order of the existing row.
+    var newRow = document.createElement('tr')
+    newRow.dataset.height = block.height
+    newRow.dataset.linkClass = tr.dataset.linkClass
+    var tds = tr.querySelectorAll('td')
+    tds.forEach((td) => {
+      let newTd = document.createElement('td')
+      let dataType = td.dataset.type
+      newTd.dataset.type = dataType
+      switch (dataType) {
+        case 'index':
+          newTd.textContent = block.windowIndex
+          break
+        case 'height':
+          newTd.textContent = block.Height
+          let link = document.createElement('a')
+          link.href = `/block/${block.height}`
+          link.textContent = block.height
+          link.classList.add(tr.dataset.linkClass)
+          newTd.appendChild(link)
+          break
+        case 'txs':
+          newTd.textContent = block.tx
+          break
+        case 'voters':
```

Comment:
Would be ideal to have just one label per datum. We already have an annoying discrepancy between the field names used in the go templates and the json api. So maybe these data attributes can stick to one of those field names?

Eg here `Voters` `votes` and `voters` are all labels used to represent the same datum.
If naming was more unified, then many cases of this switch could be handling via a default case of `newTd.textContent = block[dataType]`.

Reply:
That does clean it up a bit. Thank you.